source/common/http/http1/codec_impl.cc

```diff
@@ -614,10 +565,10 @@ Http::Status ConnectionImpl::innerDispatch(Buffer::Instance& data) {
       }
 
       total_parsed += statusor_parsed.value();
-      if (HTTP_PARSER_ERRNO(&parser_) != HPE_OK) {
+      if (parser_->getErrno() != parser_->statusToInt(ParserStatus::Success)) {
         // Parse errors trigger an exception in dispatchSlice so we are guaranteed to be paused at
         // this point.
-        ASSERT(HTTP_PARSER_ERRNO(&parser_) == HPE_PAUSED);
+        ASSERT(parser_->getErrno() == parser_->statusToInt(ParserStatus::Paused));
```

Comment:
If think this pattern is used in quite all other places. Would it make sense to add `Parser::getStatus()` method that translates parser's error code to status so that the code would look lie this:
```
ASSERT(parser_->getStatus() == ParserStatus::Paused);
``` 
I think it will make it more readable.